waiter/test/waiter/scheduler/marathon_test.clj

```diff
@@ -588,13 +588,28 @@
                                         {:service-id service-id, :service-description service-description})]
         (is (= expected actual))))))
 
+(defn- create-marathon-scheduler
+  [& {:as marathon-config}]
+  (-> (merge {:force-kill-after-ms 1000
+              :home-path-prefix "/home/path/"
+              :is-waiter-app?-fn (constantly true)
+              :marathon-api {}
+              :mesos-api {}
+              :retrieve-framework-id-fn (constantly nil)
+              :service-id->failed-instances-transient-store (atom {})
+              :service-id->kill-info-store (atom {})
+              :service-id->out-of-sync-state-store (atom {})
+              :service-id->password-fn #(str % ".password")
+              :service-id->service-description (constantly nil)
+              :sync-deployment-maintainer-atom (atom nil)}
+             marathon-config)
+      map->MarathonScheduler))
```

Comment:
Nitpick: This use of threading doesn't seem to be buying you anything. I'd either drop the threading altogether to get `(map->MarathonScheduler (merge ...))`, or lift the first arg out of the merge to get `(-> {...} (merge marathon-config) map->MarathonScheduler)`.

Reply:
Did the second :)